spec/datadog_spec.rb

```diff
@@ -397,44 +398,60 @@
       end
 
       it 'retries no more than twice' do
-        @handler.run_report_unsafe(@run_status)
+        dogs = @handler.instance_variable_get(:@dogs)
 
-        expect(a_request(:put, HOST_TAG_ENDPOINT + @node.name).with(
-          :query => { 'api_key' => @handler.config[:api_key],
-                      'application_key' => @handler.config[:application_key],
-                      'source' => 'chef' },
-          :body => hash_including(:tags => [
-            'env:hostile', 'role:highlander', 'tag:the_one_and_only'
-            ]),
-        )).to have_been_made.times(3)
+        # Check that we do have a Dogapi client
+        expect(dogs.length).to eq(1)
```

Comment:
Making two expectations in separate tests might be better following the single assert rule. :)

Reply:
(multiple occurence)